python/tk_desktop/communication_base.py

```diff
@@ -101,24 +101,20 @@ def _create_proxy(self, pipe, authkey):
         Connects to the other process's RPC server.
         """
         logger.info("Connecting to gui pipe %s" % pipe)
-        self._proxy = RPCProxy(pipe, authkey)
-        logger.debug("Connected to the proxy server.")
+        self._proxy = get_rpc_proxy_factory(pipe)(pipe, authkey)
+        logger.info(
+            "%s connected to proxy server using %s(%s)."
+            % (type(self).__name__, type(self._proxy).__name__, pipe)
+        )
 
-    def _create_server(self):
+    def _create_server(self, rpc_server_factory):
```

Comment:
That's because on the SiteCommunication side, we always use the DualRPCServer class, since we don't know how the background desktop will connect. On the ProjectCommunication side however, we'll use the http proxy or the multiprocessing proxy, depending on which is available.